plugins/generator-1.18.2/forge-1.18.2/templates/rangeditem/rangeditem.java.ftl

```diff
@@ -153,7 +157,7 @@ public class ${name}Item extends Item {
 		}
 	}
 	<#else>
-	entityarrow.pickup = AbstractArrow.Pickup.DISALLOWED;
+	entityarrow.pickup = AbstractArrow.Pickup.<#if !data.pickupProjectiles>DISALLOWED<#else>ALLOWED</#if>;

```

Comment:
If the pickup option is selected, you should instead check if the player is in creative mode and set the pickup type to `CREATIVE_ONLY`